src/goose/_logger.py

```diff
@@ -1,11 +1,59 @@
 import logging
 from pathlib import Path
+import json
+from typing import Union
+from exchange import Message, ToolResult
 
 _LOGGER_NAME = "goose"
 _LOGGER_FILE_NAME = "goose.log"
 
+_TRACE_LOGGER_NAME = "trace_logger"
+_TRACE_LOGGER_FILE_NAME = "trace.log"
+
+class TraceFilter(logging.Filter):
+    def __init__(self, toolResultOutputMaxTokens=1000):
+        super().__init__()
+        self.toolResultOutputMaxTokens = toolResultOutputMaxTokens
+
+    def filter(self, record):
+        if hasattr(record, 'trace_contents'):
+            record.msg = self.parse_trace_message(record.trace_contents)
+        return True  
+
+    def parse_trace_message(self, message: Union[str, dict, Message, ToolResult]):
+        # Custom parsing logic for trace messages
+        logMsg = ""
+        try:
+            if isinstance(message, Message):
```

Comment:
note: this is purely stylistic, feel free to ignore!

to keep this separation of concerns clearer, we might consider extracting `isinstance` into helper functions. naming is hard but something like this:

```python
def _render_dict(message: dict) -> str:
def _render_tool_result(message: ToolResult) -> str:
```

it would be easier to test and adding other message handers later on

Reply:
I like this idea, do you think these should be helper functions of TraceFilter or pulled out? (updated to functions of the class for now)

Reply:
up to you! seems more convenient for now to keep it in `TraceFilter`? easy to move later in if needed